support-frontend/assets/pages/contributions-landing/components/StripeCardForm/StripeCardForm.jsx

```diff
@@ -26,35 +27,45 @@ import { trackComponentLoad } from 'helpers/tracking/behaviour';
 /* eslint-disable react/no-unused-prop-types */
 type PropTypes = {|
   stripe: Object, // Available through the injectStripe 'Higher-Order Component'
-  onPaymentAuthorised: (PaymentAuthorisation) => Promise<PaymentResult>,
+  onPaymentAuthorised: (paymentMethodId: string) => Promise<PaymentResult>,
   paymentFailure: (paymentError: ErrorReason) => Action,
   contributionType: ContributionType,
   setCreateStripePaymentMethod: ((email: string) => void) => Action,
   setHandleStripe3DS: ((clientSecret: string) => Promise<Stripe3DSResult>) => Action,
-  paymentWaiting: (isWaiting: boolean) => Action,
+  setPaymentWaiting: (isWaiting: boolean) => Action,
+  paymentWaiting: boolean,
   setStripeCardFormComplete: (isComplete: boolean) => Action,
   checkoutFormHasBeenSubmitted: boolean,
-
+  stripeKey: string,
+  setupIntentClientSecret: string | null,
+  setSetupIntentClientSecret: (setupIntentClientSecret: string) => Action,
 |};
 
 const mapStateToProps = (state: State) => ({
   contributionType: state.page.form.contributionType,
   checkoutFormHasBeenSubmitted: state.page.form.formData.checkoutFormHasBeenSubmitted,
+  setupIntentClientSecret: state.page.form.stripeCardFormData.setupIntentClientSecret,
+  paymentWaiting: state.page.form.isWaiting,
 });
 
 const mapDispatchToProps = (dispatch: Function) => ({
-  onPaymentAuthorised:
-    (paymentAuthorisation: PaymentAuthorisation) =>
-      dispatch(onThirdPartyPaymentAuthorised(paymentAuthorisation)),
+  onPaymentAuthorised: (paymentMethodId: string) =>
```

Comment:
Could be beneficial to enforce something more concise than `string`, e.g using an enum type or similar.

Reply:
the payment method here is a Stripe api concept - https://stripe.com/docs/payments/payment-methods
The `paymentMethodId` here is effectively a string token given to us by Stripe

Reply:
@tomrf1 Makes sense.